### ISSUE TITLE
Reduce allocations in non-in place form

### ISSUE THREAD

**Issue by fgasdia** [fgasdia/Romberg.jl#2]
The non-in place forms `romberg(x,y)` and `romberg(x, y, max_steps)` are often slightly slower than the in place form because they still need to generate two `(max_steps + 1)` vectors and then make the same number of mutations as done for the in place form.

Is it possible to reduce allocations or increase speed? Maybe using MArrays or generated function because `max_steps + 1` is fairly small (`typemax(Int64)` would lead to a maximum size of 63).

Still, this package's runtime is dominated by calls to `trapz`...

**Comment by fgasdia** [comment]
This is no longer relevant after #4
